flac.go

```diff
@@ -253,7 +278,99 @@ func (stream *Stream) ParseNext() (f *frame.Frame, err error) {
 	return frame.Parse(stream.r)
 }
 
-// Seek is not implement yet.
-func (stream *Stream) Seek(offset int64, whence int) (read int64, err error) {
-	return 0, nil
+// Seek to offset where offset represents the sample number in the flac file
+//
+// TODO: return an error if sample is invalid?  Do nothing?
+//
+// sample could be invalid if:
+// whence == io.SeekEnd && sample is not negative
+// whence == io.SeekStart && sample is negative
+// whence == io.SeekCurrent and sample + current sample < 0 or > stream.Info.NSamples
+func (stream *Stream) Seek(sample int64, whence int) (read int64, err error) {
```

Comment:
praise: This is super clean, I like how easy it reads by finding a seek point and just `r.Seek(start+p.Offset)`. Great work!